mailbox/opensearch/src/test/java/org/apache/james/mailbox/opensearch/OpenSearchNoIndexBodyIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+        ReadAliasName readAliasName = new ReadAliasName(UUID.randomUUID().toString());
+        WriteAliasName writeAliasName = new WriteAliasName(UUID.randomUUID().toString());
+        indexName = new IndexName(UUID.randomUUID().toString());
+        MailboxIndexCreationUtil.prepareClient(client, readAliasName, writeAliasName, indexName,
+            openSearch.getDockerOpenSearch().configuration());
+
+        InMemoryIntegrationResources resources = InMemoryIntegrationResources.builder()
+            .preProvisionnedFakeAuthenticator()
+            .fakeAuthorizator()
+            .inVmEventBus()
+            .defaultAnnotationLimits()
+            .defaultMessageParser()
+            .listeningSearchIndex(preInstanciationStage -> new OpenSearchListeningMessageSearchIndex(
+                preInstanciationStage.getMapperFactory(),
+                ImmutableSet.of(),
+                new OpenSearchIndexer(client, writeAliasName),
+                new OpenSearchSearcher(client, new QueryConverter(new CriterionConverter()), SEARCH_SIZE, readAliasName, routingKeyFactory),
+                new MessageToOpenSearchJson(textExtractor, ZoneId.of("Europe/Paris"), IndexAttachments.YES, IndexHeaders.YES),
+                preInstanciationStage.getSessionProvider(), routingKeyFactory, messageIdFactory,
+                OpenSearchMailboxConfiguration.builder().indexBody(IndexBody.NO).build(), new RecordingMetricFactory()))
+            .noPreDeletionHooks()
+            .storeQuotaManager()
+            .build();
+
+        storeMailboxManager = resources.getMailboxManager();
+        messageSearchIndex = resources.getSearchIndex();
+
+        session = storeMailboxManager.createSystemSession(USERNAME);
+        inboxPath = MailboxPath.inbox(USERNAME);
+        storeMailboxManager.createMailbox(inboxPath, session);
+
+        StoreMessageManager inboxMessageManager = (StoreMessageManager) storeMailboxManager.getMailbox(inboxPath, session);
+        mailbox = inboxMessageManager.getMailboxEntity();
+    }
+
+    @AfterEach
+    void tearDown() throws IOException {
+        client.close();
+    }
+
+    @Test
+    void searchingByBodyContentShouldNotReturnMessageWhenNoIndexBody() throws Exception {
+        addMessage(session, inboxPath).block();
+
+        awaitForOpenSearch(QueryBuilders.matchAll().build()._toQuery(), 1L);
+
+        SearchQuery searchQuery = SearchQuery.of(SearchQuery.bodyContains("Hello"));
+
+        assertThat(messageSearchIndex.search(session, mailbox, searchQuery).collectList().block())
+            .isEmpty();
+    }
+
+    @Test
+    void searchingAllShoulReturnMessageWhenNoIndexBody() throws Exception {
+        ComposedMessageId composedMessageId = addMessage(session, inboxPath).block();
+
+        awaitForOpenSearch(QueryBuilders.matchAll().build()._toQuery(), 1L);
+
+        SearchQuery searchQuery = SearchQuery.of(SearchQuery.bodyContains("Hello"));
```

Comment:
```suggestion
        SearchQuery searchQuery = SearchQuery.of(SearchQuery.all());
```

To match the test name?